File: 2259-remove-digit-from-number-to-maximize-result/2259-remove-digit-from-number-to-maximize-result.py
class Solution:
    def removeDigit(self, s: str, digit: str) -> str:
        for i in range(len(s)-1):
            if s[i] == digit and s[i+1] > digit:
                return s[:i] + s[i+1:]
        for i in range(len(s)-1, -1, -1):
            if s[i] == digit:
                return s[:i] + s[i+1:]

File: 2259-remove-digit-from-number-to-maximize-result/2259-remove-digit-from-number-to-maximize-result.java
class Solution {
    public String removeDigit(String s, char digit) {
        for(int i = 0; i < s.length() - 1; i++){
            if(s.charAt(i) == digit && s.charAt(i+1) > digit)
                return s.substring(0, i) + s.substring(i+1, s.length());
        }
        for(int i = s.length() - 1; i >= 0; i--){
            if(s.charAt(i) == digit)
                return s.substring(0, i) + s.substring(i+1, s.length());
        }
        return "";
    }
}